lib/data.php

```diff
@@ -10,22 +10,19 @@
 
 // Call the REDCap Connect file in the main "redcap" directory; enforces permissions.
 require_once dirname(realpath(__FILE__)) . '/../../../redcap_connect.php';
+require_once(dirname(realpath(__FILE__)) . '/ReportConfig.php');
 
-/**
- * Filter the input
- */
-function filterInput($str) {
-  return filter_var($str, FILTER_SANITIZE_STRING, FILTER_FLAG_NO_ENCODE_QUOTES);
-}
+$reportConfigInstance = new \Octri\ConsortReport\ReportConfig($project_id, $module);
+$reportConfig = json_decode($reportConfigInstance->getReportConfig(), true);
 
 $returnArray = array();
-foreach ($_POST['reportIds'] as $reportId) {
-    $reportId = intval(filterInput($reportId));
-    $report = REDCap::getReport($reportId, 'json');
-    $returnArray[] = array(
-        'reportId' => $reportId,
-        'totalRecords' => count(json_decode($report, true))
-    );
+foreach ($reportConfig as $reportNode) {
+    $report = REDCap::getReport($reportNode['reportId'], 'json');
+    $returnArray[] = array_merge($reportNode, array(
```

Comment:
To reduce the response size this could be smarter. If the `strategy` is `total` it doesn't need to send the report data, and the report data could be slimmed down to only include the `bucket-by` content.